pkg/proxy/ipvs/proxier.go

```diff
@@ -900,7 +900,7 @@ func (proxier *Proxier) syncProxyRules() {
 	defer proxier.mu.Unlock()
 
 	// don't sync rules till we've received services and endpoints
-	if !proxier.endpointsSynced || !proxier.servicesSynced {
+	if !proxier.servicesSynced || !(proxier.endpointsSynced || proxier.endpointSlicesSynced) {
```

Comment:
I didn't expect to see code checking `endpointSlicesSynced` unprotected by alpha feature gate checks yet

Reply:
Since `endpointSlicesSynced` defaults to false it should have no affect here unless it's actually set. The value of this variable should not change unless the EndpointSlice feature gate has been enabled. I'd hoped that was sufficient isolation, but happy to add a feature gate check here as well if that makes more sense.

Reply:
To be safe, we should factor this out with the feature gate...

Reply:
👍 Done - feature gates added to both iptables and ipvs proxiers.

Reply:
looks good, thanks